### PR TITLE
HAWQ-1001. Bugfix and refactor ranger logic.

### PR DESCRIPTION
**Pull request by xunzhang**

### CODE REVIEW

**Comment by xunzhang** [comment]
cc @linwen @ictmalili  @zhangh43

File: src/backend/libpq/rangerrest.c [hunk 1]
```diff
@@ -273,10 +245,10 @@ json_object *create_ranger_request_json_batch(List *args)
  *   }
  */
 json_object* create_ranger_request_json(char* user, AclObjectKind kind, char* object,
-        List* actions, char* how)
+        List* actions, bool how)
```
Comment (ictmalili) [inline src/backend/libpq/rangerrest.c]: What do we pass for "how", why change it from char* to bool?

Comment (xunzhang) [inline src/backend/libpq/rangerrest.c]: https://github.com/xunzhang/incubator-hawq/blob/3568cbfb790a196e555ed067d13a7483ee872f34/src/backend/catalog/aclchk.c#L2776

Comment (paul-guo-) [inline src/backend/libpq/rangerrest.c]: Even change to bool, isn't another name better like is_*?

File: src/backend/libpq/rangerrest.c [hunk 2]
```diff
@@ -391,7 +363,7 @@ static size_t write_callback(char *contents, size_t size, size_t nitems,
         void *userp)
 {
     size_t realsize = size * nitems;
-    CURL_HANDLE curl = (struct curl_context *) userp;
+    CURL_HANDLE curl = (struct curl_context_t *) userp;
```
Comment (paul-guo-) [inline src/backend/libpq/rangerrest.c]: Usually if a type have a _t in end, that means no struct is needed.

Either "struct curl_context" or curl_context_t.

File: src/backend/libpq/rangerrest.c [hunk 3]
```diff
@@ -468,7 +440,7 @@ void call_ranger_rest(CURL_HANDLE curl_handle, char* request)
     }
     else
     {
-        elog(LOG, "%lu bytes retrieved from Ranger Restful API.",
+        elog(LOG, "%d bytes retrieved from Ranger Restful API.",
```
Comment (paul-guo-) [inline src/backend/libpq/rangerrest.c]: Why not use  size_t for size in struct?

File: src/backend/libpq/rangerrest.c (hunk 1 above)
Comment (xunzhang) [inline src/backend/libpq/rangerrest.c]: Good advice!

File: src/backend/libpq/rangerrest.c (hunk 2 above)
Comment (xunzhang) [inline src/backend/libpq/rangerrest.c]: ok, thanks.

File: src/backend/libpq/rangerrest.c (hunk 3 above)
Comment (xunzhang) [inline src/backend/libpq/rangerrest.c]: It should be, but this pull request is just a hotfix and small refactor. I think ranger codes should be largely refactored later on which is not relevant to this pull request. Thanks.

**Comment by radarwave** [comment]
+1

**Comment by xunzhang** [comment]
Merged into master.

**Pull request closed without merging**